### DISCUSSION TITLE
How to structurize multi-app projects?

### DISCUSSION THREAD

**Discussion by mmzeynalli** [aminalaee/sqladmin#527] [Q&A]
I have the project, which has three (for now) apps: auth, core, logic. Each of them has `models.py` file, and related `admin.py` file. In Django, I am kind of used to solve everything in `admin.py`, but I cannot seem to do it with FastAPI and SQLAdmin. I tried to create `admin` object in `main.py`, and import it in every apps' `admin.py` file, but it does not work. The only way of making it work is to write everything in `main.py`, which I believe, is not that effective, as code base grows. Any suggestion how can I handle this situation? Thanks beforehand.

**Comment by aminalaee** [comment]
This was discussed in other issues I think.
You can define your ModelView classes in different admin.py files and import them and register them the same way you include FastAPI routers in the `main.py` file.
Another option which was also suggested is that we can change `admin = Admin(engine=engine)` to optionally work with `admin.init_app(app)` to allow loading app later on. I think this will simplify it.
Feel free to create a PR for this, it should be straightforward.
